universal-login-relayer/lib/services/MessageHandler.ts

```diff
@@ -42,26 +42,19 @@ class MessageHandler {
     }
   }
 
-  async executeSigned(message: SignedMessage) {
-    const requiredSignatures = await getRequiredSignatures(message.from, this.wallet);
-    if (requiredSignatures > 1) {
-      const hash = await this.pendingMessages.add(message);
-      const messageStatus = await this.pendingMessages.getStatus(hash);
-      const numberOfSignatures = messageStatus.totalCollected;
-      if (await this.pendingMessages.isEnoughSignatures(hash) && numberOfSignatures !== 1) {
-        return this.executePending(hash, message);
-      }
-      return JSON.stringify(await this.pendingMessages.getStatus(hash));
-    } else {
-      return this.executor.execute(message);
+  async handleMessage(message: SignedMessage) {
+    const hash = await this.pendingMessages.add(message);
+    if (await this.pendingMessages.isEnoughSignatures(hash)) {
+      return this.executePending(hash, message);
```

Comment:
I would consider moving `if (... isEnoughSignatures(...)) {` to `PendingMessages`.
`PendingMessages` should return information to MessageHandler via callback which is connected directly to `executePending`.
Next PR?

Reply:
I consider it could be done after moving logic of function 'add' (PendingMessages -> PendingMessagesStore)

Reply:
✔️